content/blog/scales-1-3-0/index.Rmd

```diff
@@ -0,0 +1,101 @@
+---
+output: hugodown::hugo_document
+
+slug: scales-1-3-0
+title: scales 1.3.0
+date: 2023-11-07
+author: Thomas Lin Pedersen
+description: >
+    scales 1.3.0 is a minor release focusing on streamlining the API and gradual improvements 
+    of the existing utilities
+
+photo:
+  url: https://unsplash.com/photos/gold-and-silver-round-frame-magnifying-glass-j06gLuKK0GM
+  author: Elena Mozhvilo
+
+# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
+categories: [package] 
+tags: []
+---
+
+```{=html}
+<!--
+TODO:
+* [ ] Look over / edit the post's title in the yaml
+* [ ] Edit (or delete) the description; note this appears in the Twitter card
+* [ ] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
+* [ ] Find photo & update yaml metadata
+* [ ] Create `thumbnail-sq.jpg`; height and width should be equal
+* [ ] Create `thumbnail-wd.jpg`; width should be >5x height
+* [ ] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+```
+We're delighted to announce the release of [scales](https://scales.r-lib.org) 1.3.0. scales is a packages that extracts much of the scaling logic that is used in ggplot2 to a general framework, along with utility functions for e.g. formatting labels or creating color palettes.
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("scales")
+```
+
+This blog post will give a quick overview of the 1.3.0 release, which is mainly an upkeep release but does contain a few interesting tidbits.
+
+You can see a full list of changes in the [release notes](%7B%20github_release%20%7D)
+
+```{r setup}
+library(scales)
+set.seed(1)
+```
+
+## Proper support for difftime objects
+
+While scales have had rudimentary support for objects from the hms package it has not included basic support for difftime objects from base. This is now rectified with the introduction of `label_timespan()`, `breaks_timespan()`, and `transform_timespan()`. While the labels and breaks function can be used on their own, all the behavior is encapsulated in the timespan transform object which is kin to `transform_hms()`.
+
+```{r}
+library(ggplot2)
+
+events <- data.frame(
+  time = runif(30, max = 200),
+  magnitude = rnorm(30) + 2
+)
+
+ggplot(events) + 
+  geom_point(aes(time, y = 0, size = magnitude), position = position_jitter(height = 0)) + 
+  scale_x_continuous(trans = transform_timespan(), limits = c(0, NA))
+```
+
+As we can see the timespan transform by default understand numeric data as seconds elapsed. Further it identifies that for this range, adding breaks for minutes makes most sense.
+
+If we had recorded time as hours rather than seconds, we can inform the transformer of this:
```

Comment:
I think it would be better to illustrate this by setting the `units` attribute.

Reply:
The point was mostly to show that it is adaptable to how a numeric should be interpreted. But maybe it is better to focus on the difftime object directly